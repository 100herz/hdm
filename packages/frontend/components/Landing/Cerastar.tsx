import styled from '@emotion/styled'
import dynamic from 'next/dynamic'

export default function Cerastar({ className = '' }: React.HTMLAttributes<HTMLElement>) {
  const Pdf = dynamic(() => import('@/components/Utils/Pdf'), { ssr: false })

  return (
    <section
      id="cerastar"
      className={`container relative lg:grid lg:grid-cols-2 lg:gap-32 mt-32 pt-20 pb-24 lg:mt-24 ${className}`}
    >
      <BackgroundImage />

      <header>
        <div className="lg:ml-32">
          <h2 className="text-4xl font-bold lg:text-5xl">Cerastar</h2>
          <p className="mt-4 text-xl leading-snug lg:text-2xl">
            Der neue Star unter <br />
            den Bodenbelägen
          </p>
        </div>

        <picture className="block mt-16 -mx-6 lg:-ml-20 lg:-mr-8">
          <source srcSet="/images/hdm-cerastar.webp" type="image/webp" />
          <source srcSet="/images/hdm-cerastar.png" type="image/png" />
          <img src="/images/hdm-cerastar.png" alt="Cerastar - der neue Star unten den Bodenbelägen" />
        </picture>
      </header>

      <div className="lg:mt-32 lg:pr-16">
        <h3 className="mt-12 font-bold lg:text-lg">
          Cerastar heißt der neue Designbodenbelag mit keramischer Nanocoat Oberflächenversiegelung.
        </h3>
        <p className="mt-8 lg:text-lg">
          1,2 Mio. feinste Keramikpartikel pro Quadratzentimeter machen die Oberfläche von Cerastar – gegenüber
          Oberflächen vergleichbarer Bodenbeläge – bis zu dreimal kratzfester und strapazierfähiger.
        </p>
        <p className="mt-8 lg:text-lg">
          So entsteht eine äußerst abriebfeste Schutzschicht, die deutlich resistenter gegen Mikrokratzer ist. Zugleich
          verleiht die hochtransparente Versiegelung den Dekoren eine einzigartige brillante Farbtiefe und eleganten
          Glanz.
        </p>
        <p className="mt-8 lg:text-lg">
          Weitere Informationen finden Sie unter{' '}
          <a className="underline text-hdm hover:text-hdm-dark" href="https://cerastar.eu">
            cerastar.eu
          </a>
          .
        </p>

        <div className="mt-16 text-center lg:text-left lg:mt-24">
          <Pdf title="Cerastar Katalog" filename="/documents/hdm-cerastar.pdf" />
        </div>
      </div>
    </section>
  )
}

const BackgroundImage = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: -2;
  overflow-x: hidden;
  background-color: #f8efef;

  &::before {
    position: absolute;
    top: 3rem;
    left: 45%;
    z-index: -1;
    width: 581px;
    height: 409px;
    background: url(/images/hdm-cerastar-muster.png) no-repeat;
    opacity: 0.7;
    content: '';
  }

  @media (min-width: 1024px) {
    &::before {
      top: 5rem;
    }
  }
`
