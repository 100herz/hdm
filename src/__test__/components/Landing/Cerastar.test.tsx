import { ShallowWrapper, shallow } from "enzyme"
import toJSON from "enzyme-to-json"

import Cerastar from "@/components/Landing/Cerastar"

describe("Cerastar Testings", () => {
  let wrapper: ShallowWrapper

  beforeEach(() => {
    wrapper = shallow(<Cerastar />)
  })

  test("matches snapshot", () => {
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
