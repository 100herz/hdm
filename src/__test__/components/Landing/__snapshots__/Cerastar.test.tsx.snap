// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cerastar Testings matches snapshot 1`] = `
<section
  className="container relative lg:grid lg:grid-cols-2 lg:gap-32 mt-32 pt-20 pb-24 lg:mt-24 "
  id="cerastar"
>
  <Cerastar__BackgroundImage />
  <header>
    <div
      className="lg:ml-32"
    >
      <h2
        className="text-4xl font-bold lg:text-5xl"
      >
        Cerastar
      </h2>
      <p
        className="mt-4 text-xl leading-snug lg:text-2xl"
      >
        Der neue Star unter 
        <br />
        den Bodenbelägen
      </p>
    </div>
    <picture
      className="block mt-16 -mx-6 lg:-ml-20 lg:-mr-8"
    >
      <source
        srcSet="/images/hdm-cerastar.webp"
        type="image/webp"
      />
      <source
        srcSet="/images/hdm-cerastar.png"
        type="image/png"
      />
      <img
        alt="Cerastar - der neue Star unten den Bodenbelägen"
        src="/images/hdm-cerastar.png"
      />
    </picture>
  </header>
  <div
    className="lg:mt-32 lg:pr-16"
  >
    <h3
      className="mt-12 font-bold lg:text-lg"
    >
      Cerastar heißt der neue Designbodenbelag mit keramischer Nanocoat Oberflächenversiegelung.
    </h3>
    <p
      className="mt-8 lg:text-lg"
    >
      1,2 Mio. feinste Keramikpartikel pro Quadratzentimeter machen die Oberfläche von Cerastar – gegenüber Oberflächen vergleichbarer Bodenbeläge – bis zu dreimal kratzfester und strapazierfähiger.
    </p>
    <p
      className="mt-8 lg:text-lg"
    >
      So entsteht eine äußerst abriebfeste Schutzschicht, die deutlich resistenter gegen Mikrokratzer ist. Zugleich verleiht die hochtransparente Versiegelung den Dekoren eine einzigartige brillante Farbtiefe und eleganten Glanz.
    </p>
    <p
      className="mt-8 lg:text-lg"
    >
      Weitere Informationen finden Sie unter
       
      <a
        className="underline text-hdm hover:text-hdm-dark"
        href="https://cerastar.eu"
      >
        cerastar.eu
      </a>
      .
    </p>
    <div
      className="mt-16 text-center lg:text-left lg:mt-24"
    >
      <ForwardRef(LoadableComponent)
        filename="/documents/hdm-cerastar.pdf"
        title="Cerastar Katalog"
      />
    </div>
  </div>
</section>
`;
